### DISCUSSION TITLE
Dynamic routes 404 in prod when refreshed/linked directly

### DISCUSSION THREAD

**Discussion by dethstrobe** [vercel/next.js#21235] [Help] [answered]
[Link to the Repo](https://github.com/HeyOmae/OMA3)
[Link to prod site](https://oma3.vercel.app/)

## Descriptions
This is a web app to generate a character for a role playing game. The data is stored on the client's indexDB. When a page is refreshed or a user enters the url manually, the site 404s when trying to access the page. Navigating to the route with NextJS Links from the base url with links works as expected.

### Steps to reproduce
1. Go to https://oma3.vercel.app/
2. click "Create Runner" button
3. refresh page

#### Expected
Page to reload

#### Actual
Page 404s

### Things I've tried
* `npm run dev` while slow, works as expected
* `npm run build && npm start` works as expected
* Added a `vercel.json` to redirect to the root, but it only loaded the base app and not the linked route
* Tried to add `getStaticProps` and `getStaticPaths` but from what little experimentation i've tried it didn't work and I don't really like it, as I'm not doing any dynamic http requests.

---

I've been stuck on this for a few weeks and trying a lot of different things to see if I can get around it. But I can't seem to figure out why the routing doesn't seem to work on vercel. Any help would be much appreciated.

---
**EDIT** - just to make it easier on myself. [This issue](https://github.com/vercel/vercel/issues/3294#issuecomment-817321148) might prove fruitful later, but does nothing to help me now.

**Comment by joshkay** [comment]
I am also running into this issue.  Any luck with a solution?

**Reply by dethstrobe** [reply]
Not yet sadly.

**Comment by dethstrobe** [comment, marked as answer]
I'm not sure why, but just simply [disabling static export in my config file](https://github.com/HeyOmae/OMA3/commit/7b255c3133786a42e36fcc6f2349c7ca92806067) "fixed" this for me.

I feel like this is wrong, because static export should support dynamic routing. But whatever, I'm not going to split hairs over it, and it's not like I need to hyper optimize for performance.
